src/components/Editor.js

```diff
@@ -102,10 +101,27 @@ const Editor = React.createClass({
 
   displayName: "Editor",
 
+  getInitialState() {
+    return {
+      query: "",
```

Comment:
This state had to be moved up into the editor as getting the query from `ctx.cm.getSelection()` would clear regex/whole word searches.